src/test/java/nextstep/subway/domain/SectionsTest.java

```diff
@@ -81,135 +154,160 @@ void setUp() {
     }
 
     /**
-     * Given 새로운 역을 추가 후
-     * When 기존 구간 상행역에 구간을 추가 요청 시
-     * Then 추가가 된다.
+     * When 구간에 주어진 역을 조회 요청 시
+     * Then 해당하는 구간을 찾을 수 있다
      */
-    @DisplayName("새로운 역을 추가 후 기존 구간 상행역에 구간을 추가 요청 시 추가가 된다")
+    @DisplayName("구간에 주어진 역을 조회 요청 시 해당하는 구간을 찾을 수 있다")
     @Test
-    void 새로운_역을_추가_후_기존_구간_상행역에_구간을_추가_요청_시_추가가_된다() {
-        // Given
-        Station 논현역 = new Station("논현역");
-
+    void 구간에_주어진_역을_조회_요청_시_해당하는_구간을_찾을_수_있다() {
         // When
-        givenSections.addSection(신분당선, 논현역, 강남역, 7);
+        Section sectionByGangNam = givenSections.findSectionByStation(강남역);
+        Section sectionByYangJaeStation = givenSections.findSectionByStation(양재역);
 
         // Then
-        List<Section> sections = givenSections.getSections();
-        assertThat(sections.get(0).getUpStation()).isEqualTo(강남역);
-        assertThat(sections.get(0).getDownStation()).isEqualTo(양재역);
-        assertThat(sections.get(0).getDistance()).isEqualTo(10);
-        assertThat(sections.get(1).getUpStation()).isEqualTo(논현역);
-        assertThat(sections.get(1).getDownStation()).isEqualTo(강남역);
-        assertThat(sections.get(1).getDistance()).isEqualTo(7);
+        assertThat(sectionByGangNam.getUpStation()).isEqualTo(강남역);
+        assertThat(sectionByGangNam.getDownStation()).isEqualTo(양재역);
```

Comment:
다음과 같이 검증해 볼 수도 있을 것 같아요 😄 
```suggestion
        assertThat(givenSection).isEqualTo(sectionByGangNam);
```